tests/providers/microsoft/azure/hooks/test_adx.py

```diff
@@ -0,0 +1,172 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+import json
+import unittest
+from unittest import mock
+
+from azure.kusto.data.request import ClientRequestProperties, KustoClient, KustoConnectionStringBuilder
+
+from airflow.exceptions import AirflowException
+from airflow.models import Connection
+from airflow.providers.microsoft.azure.hooks.adx import AzureDataExplorerHook
+from airflow.utils import db
+
+
+class TestAzureDataExplorerHook(unittest.TestCase):
+    def test_conn_missing_method(self):
+        db.merge_conn(
+            Connection(conn_id='adx_test_key_missing_method',
+                       conn_type='azure_data_explorer',
+                       login='client_id',
+                       password='client secret',
+                       host='https://help.kusto.windows.net',
+                       extra=json.dumps({})))
```

Comment:
Could you make a tearDown function that removes this/all these extra connections you have set up?

(To make that tear down easier it might be easier if you use the same conn_id for each test, but with differing fields/settings etc.)

Reply:
```python
self.session.query(Connection).filter(Connection.conn_id.like_("adx_text_%")).delete()
```

should do it.

Reply:
Done.